examples/sparse/gcnii.py

```diff
@@ -0,0 +1,179 @@
+"""
+[Simple and Deep Graph Convolutional Networks]
+(https://arxiv.org/abs/2007.02133)
+"""
+
+import argparse
+import math
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from dgl.data import CoraGraphDataset
+from dgl.mock_sparse import create_from_coo, diag, identity, spmm
+from torch.nn.parameter import Parameter
+from torch.optim import Adam
+
+
+class GCNIIConvolution(nn.Module):
+    def __init__(self, in_size, out_size):
+        super().__init__()
+        self.out_size = out_size
+        self.weight = Parameter(torch.FloatTensor(in_size, out_size))
+        self.reset_parameters()
+
+    ############################################################################
+    # (HIGHLIGHT) Take the advantage of DGL sparse APIs to implement the GCNII
+    # forward process.
+    ############################################################################
+    def forward(self, A_norm, H, H0, lamda, alpha, l):
+        beta = math.log(lamda / l + 1)
+        H = spmm(A_norm, H)
+        support = (1 - alpha) * H + alpha * H0
+        H = (1 - beta) * support + beta * support @ self.weight
+        return H
+
+    def reset_parameters(self):
+        std = 1 / math.sqrt(self.out_size)
+        self.weight.data.uniform_(-std, std)
+
+
+class GCNII(nn.Module):
+    def __init__(
+        self,
+        in_size,
+        out_size,
+        hidden_size,
+        n_layers,
+        lamda,
+        alpha,
+        dropout=0.5,
+    ):
+        super().__init__()
+        self.hidden_size = hidden_size
+        self.n_layers = n_layers
+        self.lamda = lamda
+        self.alpha = alpha
+
+        self.FC_layers = nn.ModuleList()
+        self.FC_layers.append(nn.Linear(in_size, hidden_size))
+        self.FC_layers.append(nn.Linear(hidden_size, out_size))
+
+        self.CONV_layers = nn.ModuleList()
+        for _ in range(n_layers):
+            self.CONV_layers.append(GCNIIConvolution(hidden_size, hidden_size))
+        self.activation = nn.ReLU()
+        self.dropout = dropout
+
+    def forward(self, A_norm, feature):
+        H = feature
+        H = F.dropout(H, self.dropout)
```

Comment:
```suggestion
        H = F.dropout(H, self.dropout, training = self.training)
```
to handle the different behavior during `train` and `eval`.

Reply:
add the  training = self.training